### PR TITLE
CM-1178 Common owner's contributions

### LINKED ISSUE #1178: Implement immediate contribution when creating a common
When a common is created, if the owner states that people need to donate money when joining - the owner should contribute money himself first, we need to implement this functionality

### PR DESCRIPTION
**Pull request by SobolevYuri**
supposed flow:

- user's trying to create a common with a condition of minimum **contribution amount** for joining members (**zeroContribution** req.body param is set to false or absent);
- BE verifies the request params to have them suitable for the describing flow, then save a new common in the collection with an **active** field temporary set to _false_ and send the newly created common back to the FE;
- FE receives a response with the common record, check the **active** field and if its _false_ - initiate an 'immediate contribution' scenario for this common using the corresponding endpoint (and passing a needed params got from the _common_ itself);
- then the known "immediate contributions" flow runs (if successfully - common's **active** field will be set to _true_).

### CODE REVIEW

**Comment by andreymikhadyuk** [comment]
@SobolevYuri @ariel-dao I am not sure what will be if we create a common with inactive state and user will not enter his card details and just close the modal. Right now on web as I know he won’t be able to run payment again. @roienatan Please correct me if I am wrong.
Maybe we will need to add here immediate contribution feature🤷.

**Comment by roienatan** [comment]
> @SobolevYuri @ariel-dao I am not sure what will be if we create a common with inactive state and user will not enter his card details and just close the modal. Right now on web as I know he won’t be able to run payment again. @roienatan Please correct me if I am wrong.
> 
> Maybe we will need to add here immediate contribution feature🤷.

@SobolevYuri @andreymikhadyuk @ariel-dao  Yes that's true, if the user just closes the modal the common will stay `inactive` and the FE won't fetch it. It's a bit problematic. Let's talk about that.

**Comment by ariel-dao** [comment]
@SobolevYuri @andreymikhadyuk any updates about this?

**Comment by SobolevYuri** [comment]
UPDATED: redundant *founderContributionAmount* and *founderContributionType* were removed

File: functions/src/shared/interfaces/ICommonMetadata.ts [hunk 1]
```diff
@@ -31,5 +31,4 @@ export interface ICommonMetadata {
      * Will this common appear in the search results page
      */
     searchable: boolean;
-
```
Comment (SobolevYuri) [inline functions/src/shared/interfaces/ICommonMetadata.ts:34]: accidental prettier changes, nevermind

File: functions/src/shared/schemas/createCommonDataValidationScheme.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const createCommonDataValidationScheme = yup.object({
     description: yup.string(),
   
     contributionAmount: yup.number().min(0).required(),
-
```
Comment (SobolevYuri) [inline functions/src/shared/schemas/createCommonDataValidationScheme.ts]: these are, too

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**